### ISSUE TITLE
Consider using yarn instead of npm

### ISSUE THREAD

**Issue by adnelson** [adnelson/nixfromnpm#113]
I've been consistently frustrated by npm since the beginning of this project due to its obtuse operation, lack of offline support and mysterious failure conditions. Yarn is in my experience much better. It has a `--offline` flag which is pretty nice. It's meant to build from a cache and be as fast as possible. Perhaps most importantly it has reasonably decent error reporting when it fails. 

I'm not sure how much work would be required to get it to work with nix, but I think it should be doable. It has no dependencies, so it should be easy to make a custom derivation for it without needing to hack together a zipped up autogenerated nix expression like back in the npm3 days.

I might take a stab at this and at least get a sense of how feasible it would be.

**Comment by ixmatus** [comment]
I think this is a good idea and was thinking it last night. I've used the yarn2nix project in the past and my experience with it was:

- yarn is much better suited for use with nix
- generating a package set from the yarn.lock file is far more reliable than the package.json
- yarn2nix had immature nix build features compared to this project

Ultimately, I could build a large, non-trivial node project involving a giant package set, purescript code, etc. with nixfromnpm and not yarn2nix. So the flexibility you have built into this project is important to maintain.

So, yes, I think using yarn is a good idea and adding a feature to generate packages based on the lock file would be useful too.

**Comment by adnelson** [comment]
So I spent a little time on this today, and was somewhat disappointed. I couldn't figure out how to construct a faux yarn cache directory which yarn would discover packages in, which seems to be the only way to get yarn to use the `--offline` option, as even if a dependency is satisfied in `node_modules` it will still look in its cache directory. Not sure if there's a combination of options which would disable this behavior.

I put a pin in that project and instead went another direction, which I just made a PR for...
